zerver/lib/response.py

```diff
@@ -1,11 +1,35 @@
-from typing import Any, List, Mapping, Optional
+from typing import Any, Dict, Iterator, List, Mapping, Optional
 
 import orjson
 from django.http import HttpRequest, HttpResponse, HttpResponseNotAllowed
 
 from zerver.lib.exceptions import JsonableError, UnauthorizedError
 
 
+class JsonDataResponse(HttpResponse):
+    def __init__(self, data: Dict[str, Any] = {}, *, content_type: str, status: int) -> None:
+        super().__init__(content_type=content_type, status=status)
+        self.data = data
+
+    @property
+    def content(self) -> bytes:
+        # Because we don't pass a default handler, OPT_PASSTHROUGH_DATETIME
+        # actually causes orjson to raise a TypeError on datetime objects. This
+        # helps us avoid relying on the particular serialization used by orjson.
+        serialized_data = orjson.dumps(
+            self.data,
+            option=orjson.OPT_APPEND_NEWLINE | orjson.OPT_PASSTHROUGH_DATETIME,
+        )
+        return serialized_data
+
+    @content.setter
+    def content(self, value: Any) -> None:
+        pass
```

Comment:
What's the story with this? I assume it's not expected to be called; should it raise an `AssertionError` instead?

Reply:
Reworked the JsonDataResponse class so that it's more like Django's TemplateResponse, which means that this setter for the content property is no longer passed. Added a comment for the new implementation of this setter.